graylog2-web-interface/src/components/configurations/EventsConfig.jsx

```diff
@@ -0,0 +1,183 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import createReactClass from 'create-react-class';
+import { Button, FormGroup, HelpBlock } from 'react-bootstrap';
+import { BootstrapModalForm } from 'components/bootstrap';
+import { IfPermitted, TimeUnitInput } from 'components/common';
+import ObjectUtils from 'util/ObjectUtils';
+import FormUtils from 'util/FormsUtils';
+import Input from 'components/bootstrap/Input';
+import { extractDurationAndUnit } from 'components/common/TimeUnitInput';
+import lodash from 'lodash';
+import moment from 'moment';
+
+const TIME_UNITS = ['HOURS', 'MINUTES', 'SECONDS'];
+
+const EventsConfig = createReactClass({
+  displayName: 'EventsConfig',
+
+  propTypes: {
+    config: PropTypes.shape({
+      events_search_timeout: PropTypes.number,
+      events_notification_retry_period: PropTypes.number,
+      events_notification_default_backlog: PropTypes.number,
+    }),
+    updateConfig: PropTypes.func.isRequired,
+  },
+
+  getDefaultProps() {
+    return {
+      config: {
+        events_search_timeout: 60000,
+        events_notification_retry_period: 300000,
+        events_notification_default_backlog: 50,
+      },
+    };
+  },
+
+  getInitialState() {
+    const { config } = this.props;
+    return {
+      config: ObjectUtils.clone(config),
+    };
+  },
+
+  componentWillReceiveProps(newProps) {
+    this.setState({ config: ObjectUtils.clone(newProps.config) });
+  },
+
+  _openModal() {
+    this.modal.open();
+  },
+
+  _closeModal() {
+    this.modal.close();
+  },
+
+  _resetConfig() {
+    // Reset to initial state when the modal is closed without saving.
+    this.setState(this.getInitialState());
+  },
+
+  _saveConfig() {
+    const { updateConfig } = this.props;
+    const { config } = this.state;
+    updateConfig(config).then(() => {
+      this._closeModal();
+    });
+  },
+
+  _propagateChanges(key, value) {
+    const { config } = this.state;
+    const nextConfig = lodash.cloneDeep(config);
+    nextConfig[key] = value;
+    this.setState({ config: nextConfig });
+  },
+
+  _onSearchTimeoutUpdate(nextValue, nextUnit, enabled) {
+    const durationInMs = enabled ? moment.duration(nextValue, nextUnit).asMilliseconds() : 0;
+    if (this._searchTimeoutValidator(durationInMs)) {
+      this._propagateChanges('events_search_timeout', durationInMs);
+    }
+  },
+
+  _onRetryPeriodUpdate(nextValue, nextUnit, enabled) {
+    const durationInMs = enabled ? moment.duration(nextValue, nextUnit).asMilliseconds() : 0;
+    if (this._notificationsRetryValidator(durationInMs)) {
+      this._propagateChanges('events_notification_retry_period', durationInMs);
+    }
+  },
+
+  _searchTimeoutValidator(milliseconds) {
+    return milliseconds >= 1000;
+  },
+
+  _notificationsRetryValidator(milliseconds) {
+    return milliseconds >= 0;
+  },
+
+  _onBacklogUpdate(key) {
```

Comment:
This method seems to complex for what it does. Since it's only taking care of the backlog, I would rewrite it as:

```
  _onBacklogUpdate(event) {
    const value = FormUtils.getValueFromInput(event.target);
    this._propagateChanges('events_notification_default_backlog', value);
  },
```

In the component using it, you can pass a reference to the function instead of calling it directly.